src/main/java/com/programmers/springweekly/service/VoucherService.java

```diff
@@ -2,25 +2,59 @@
 
 import com.programmers.springweekly.domain.voucher.Voucher;
 import com.programmers.springweekly.domain.voucher.VoucherFactory;
-import com.programmers.springweekly.domain.voucher.VoucherType;
+import com.programmers.springweekly.dto.voucher.request.VoucherCreateRequest;
+import com.programmers.springweekly.dto.voucher.request.VoucherUpdateRequest;
+import com.programmers.springweekly.dto.voucher.response.VoucherListResponse;
+import com.programmers.springweekly.dto.voucher.response.VoucherResponse;
 import com.programmers.springweekly.repository.voucher.VoucherRepository;
-import java.util.Map;
+import java.util.List;
 import java.util.UUID;
-import lombok.AllArgsConstructor;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 @Service
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class VoucherService {
 
     private final VoucherRepository voucherRepository;
 
-    public void saveVoucher(VoucherType voucherType, String discount) {
-        Voucher voucher = VoucherFactory.createVoucher(voucherType, discount);
+    public void save(VoucherCreateRequest voucherCreateRequest) {
+        Voucher voucher = VoucherFactory.createVoucher(
+                UUID.randomUUID(),
+                voucherCreateRequest.getVoucherType(),
+                voucherCreateRequest.getDiscountAmount()
+        );
+
         voucherRepository.save(voucher);
     }
 
-    public Map<UUID, Voucher> findVoucherAll() {
-        return voucherRepository.getList();
+    public void update(VoucherUpdateRequest voucherUpdateRequest) {
+        Voucher voucher = VoucherFactory.createVoucher(
+                voucherUpdateRequest.getVoucherId(),
+                voucherUpdateRequest.getVoucherType(),
+                voucherUpdateRequest.getDiscountAmount()
+        );
+
+        voucherRepository.update(voucher);
+    }
+
+    public VoucherResponse findById(UUID voucherId) {
+        Voucher voucher = voucherRepository.findById(voucherId);
+
+        return new VoucherResponse(voucher);
+    }
+
+    public VoucherListResponse findAll() {
+        List<Voucher> voucherList = voucherRepository.findAll();
+        return new VoucherListResponse(voucherList.stream().map(VoucherResponse::new).collect(Collectors.toList()));
```

Comment:
java 17에 추가된 컬렉션 문법을 씁시다!
```suggestion
        return new VoucherListResponse(voucherList.stream().map(VoucherResponse::new).toList());
```

Reply:
헉 소나린트에 나왔었는데 제가 못봤었나보네요!! 감사합니다!! 수정했습니다!